src/hooks/useFormWithErrors.js

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect } from 'react';
+import * as yup from 'yup';
+
+export const useFormWithErrors = (
```

Comment:
potential for adding better documentation, readme with examples, and/or use of `propTypes` to fool-proof it??

I miss typescript